### ISSUE TITLE
How can I compile the browser?

### ISSUE THREAD

**Issue by Tre-brock** [mullvad/mullvad-browser#218]
I read the guide and it was not working because all of the commands are for building tor, even if I cloned and CD into the mullvad folder and used those commands it still didn't work. I also tried running ./mach build and at first it was working but then just gave me tons of errors. Could someone help me with this. Also is there a way I could build it as a .deb?

**Comment by jbjorkang** [comment]
This [repository](https://gitlab.torproject.org/tpo/applications/tor-browser-build/) is used by us at Mullvad to build the browser to verify it reproducibly. I cannot speak for how Tor produce the browser exactly, perhaps that method is slightly different.

This part of the Makefile targets for Mullvad browser: https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/blob/main/Makefile?ref_type=heads#L429

TL;DR `make mullvadbrowser-release`

**Comment by Tre-brock** [comment]
After I build it where would the compiled files be? (I'm using Linux)

**Comment by pospeselr** [comment]
@Tre-brock we have some documentaiton for `tor-browser-build` here:
- https://gitlab.torproject.org/tpo/applications/team/-/wikis/Development-Information/Tor-Browser/Building

To build the stable channel, you'll want to checkout `maint-13.0` and run:

`make mullvad-browser-release`

The builds will be found in `/mullvadbrowser/release/unsigned/${version}`

`main` is always the alpha branch which gets forked to `maint-${version}` on major version releases (and has a separate make target).

You can verify your build outputs match by comparing the sha256sums.txt file in the output directory against our own dev's hashes. We track each release in a `Release Prep` labeled issue on our gitlab. All these current open issues can be found here:

- https://gitlab.torproject.org/tpo/applications/tor-browser-build/-/issues/?sort=updated_desc&state=opened&label_name%5B%5D=Release%20Prep&first_page_size=20

Builders will post their hashes and we verify matching before signing and publishing, and you can follow our process in these issues.

One quick note is that signed release's hashes for Windows and macOS will not match due to code-signing. One of these platforms support stripping signatures to compare the builds (I forget which).  We have issues open for stripping signatures for end-user verification for the other platform but haven't gotten to it yet.
